docs/how_to_guide/convolve_large_arrays.md

```diff
@@ -0,0 +1,62 @@
+---
+jupytext:
+  text_representation:
+    extension: .md
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.16.4
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Convolve Large Arrays on the GPU
+
+Operations that can be vectorized—such as convolving multiple arrays—can benefit significantly from GPU acceleration. However, when the input arrays are large, full vectorization may cause excessive memory allocation on the GPU, potentially leading to runtime errors or causing the operation to fall back to CPU execution.
+
+By default, NeMoS vectorizes convolutions over all dimensions except the sample axis. On large arrays, this default behavior may exceed the GPU’s memory capacity. To mitigate this, you can control the memory footprint by specifying batch sizes for the convolution along the following dimensions:
+
+- **`batch_size_channels`**: batches the operation over time series channels.
+
+- **`batch_size_basis`**: batches over basis kernels.
+
+- **`batch_size_samples`**: performs the convolution in sliding windows over time, with the given sample size per batch.
+
+Use these keyword arguments inside `conv_kwargs` when initializing a convolutional basis to enable batched processing.
+
+<figure markdown>
+<!-- note that the src here has an extra ../ compared to other images, necessary when specifying path directly in html -->
+<img src="../_static/convolve_batching_scheme.svg" style="width: 100%", alt="Batched dimensions scheme."/>
+<figcaption>Schematic of the batched dimensions.</figcaption>
+</figure>
+
+## Example
+
+```{code-cell} ipython3
```

Comment:
might be useful to show memory/time performance with a couple different choices for `batch_size_dict`